quick-start/src/test/java/com/marklogic/quickstart/integrationtests/EndToEndAPITest.java

```diff
@@ -0,0 +1,350 @@
+		loginResponse = utils.doLogin();
+		
+		assertTrue(!loginResponse.isEmpty());
+		logger.info("Login Test Passed");
+		
+		// Clear all the documents in the databases
+		clearDbResponse = utils.clearAllDatabases();
+		
+		assertEquals(200, clearDbResponse.statusCode());
+		logger.info("Clear DB's test passed");
+		
+		// Test api's to check the database names and the number of documents available in each database
+		statsResponse = utils.getDocStats();
+		statsJson = statsResponse.jsonPath();
+		
+		assertEquals(200, statsResponse.statusCode());
+		assertEquals(0, statsJson.getInt("stagingCount"));
+		assertEquals(0, statsJson.getInt("finalCount"));
+		assertEquals(0, statsJson.getInt("jobCount"));
+		assertEquals(0, statsJson.getInt("traceCount"));
+		assertEquals("data-hub-STAGING", statsJson.get("stagingDb"));
+		assertEquals("data-hub-FINAL", statsJson.get("finalDb"));
+		assertEquals("data-hub-JOBS", statsJson.get("jobDb"));
+		assertEquals("data-hub-JOBS", statsJson.get("traceDb"));
+		logger.info("Stats Test Passed");
+		
+		// Test api's to create new entities
+		createEntityJsonBody = getResource("integration-test-data/create-entity-model.json");
+		createEntityJsonObj = new JSONObject(createEntityJsonBody);
+		entityName = createEntityJsonObj.getJSONObject("info").getString("title");
+		createEntityResponse = utils.createEntity(createEntityJsonBody, entityName);
+		responseEntityTitle = createEntityResponse.jsonPath().getString("info.title");
+		
+		assertEquals(200, createEntityResponse.statusCode());
+		assertTrue(responseEntityTitle.equals(entityName));
+		logger.info("Create Entities Test Passed");
+		
+		// Get the list of entities and verify if entities are created
+		getEnitiesResponse = utils.getAllEntities();
+		getEntitiesJson = getEnitiesResponse.jsonPath();
+		entitiesList = getEntitiesJson.getList("", EntityModel.class);
+		responseEntityTitle = entitiesList.get(0).getInfo().getTitle();
+		
+		assertEquals(200, getEnitiesResponse.statusCode());
+		assertEquals(1, entitiesList.size());
+		assertTrue(responseEntityTitle.equals(entityName));
+		logger.info("Verify Entities Test Passed");
+		
+		// Test API's for creation on input flows.
+		flowType = "INPUT";
+		createInputFlowResponse = utils.createFlow(entityName, flowType, DataFormat.JSON, 
+				CodeFormat.JAVASCRIPT, false, null);
+		createInputFlowJson = createInputFlowResponse.jsonPath();
+		respFlowName = createInputFlowJson.getString("flowName");
+		respEntityName = createInputFlowJson.getString("entityName");
+		respFlowType = createInputFlowJson.getString("flowType");
+		pluginDir = createInputFlowJson.getString("plugins.pluginPath");
+		
+		assertEquals(200, createInputFlowResponse.statusCode());
+		assertTrue(utils.getFlowName().equals(respFlowName));
+		assertTrue(entityName.equals(respEntityName));
+		assertTrue(flowType.equals(respFlowType));
+		assertTrue(pluginDir.contains("/"+respFlowName+"/content.sjs"));
```

Comment:
this also fails on windows

Reply:
Removed the assert for verifying the existence of plugins directory as the pluginDir results an array of Strings. The input flow is tested by checking the FlowName and EntityName and the Http Status.